docs/eslint/infinite-query-property-order.md

```diff
@@ -0,0 +1,62 @@
+---
+id: infinite-query-property-order
+title: Ensure correct order of inference sensitive properties for `useInfiniteQuery` and `infiniteQueryOptions`
+---
+
+For the following functions, the property order of the passed in object matters due to type inference:
+
+- `useInfiniteQuery`
+- `infiniteQueryOptions`
```

Comment:
we also need to add `useSuspenseInfiniteQuery` please :)

Reply:
done